jobs/batch_jobs/cron_jobs.py

```diff
@@ -0,0 +1,222 @@
+                create the user stats model.
+
+        Returns:
+            UserStatsModel. The created user stats model.
+        """
+        with datastore_services.get_ndb_context():
+            user_stats_model = (
+                user_models.UserStatsModel(id=user_settings_model.id))
+        user_stats_model.update_timestamps()
+        return user_stats_model
+
+    @staticmethod
+    def _update_weekly_creator_stats(
+            user_stats_models: user_models.UserStatsModel
+    ) -> user_models.UserStatsModel:
+        """Updates weekly dashboard stats with the current values.
+
+        Args:
+            user_stats_models: UserStatsModel. Model for which to update
+                the weekly dashboard stats.
+
+        Returns:
+            UserStatsModel. The updated user stats model.
+        """
+        model = cast(
+            user_models.UserStatsModel,
+            job_utils.clone_model(user_stats_models) # type: ignore[no-untyped-call]
+        )
+        schema_version = model.schema_version
+
+        if schema_version != feconf.CURRENT_DASHBOARD_STATS_SCHEMA_VERSION:
+            user_services.migrate_dashboard_stats_to_latest_schema(model) # type: ignore[no-untyped-call]
+
+        weekly_creator_stats = {
+            user_services.get_current_date_as_string(): { # type: ignore[no-untyped-call]
+                'num_ratings': model.num_ratings or 0,
+                'average_ratings': model.average_ratings,
+                'total_plays': model.total_plays or 0
+            }
+        }
+        model.weekly_creator_stats_list.append(weekly_creator_stats)
+        model.update_timestamps()
+        return model
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        user_settings_models = (
+            self.pipeline
+            | 'Get all UserSettingsModels' >> (
+                ndb_io.GetModels(user_models.UserSettingsModel.get_all())) # type: ignore[no-untyped-call]
+        )
+
+        old_user_stats_models = (
+            self.pipeline
+            | 'Get all UserStatsModels' >> (
+                ndb_io.GetModels(user_models.UserStatsModel.get_all())) # type: ignore[no-untyped-call]
+        )
+
+        # Creates UserStatsModels if it does not exists.
+        new_user_stats_models = (
+            (user_settings_models, old_user_stats_models)
+            | 'Merge models' >> beam.Flatten()
+            # Returns a PCollection of
+            # (model.id, (user_settings_models, user_stats_models)) or
+            # (model.id, (user_settings_models,)).
+            | 'Group models with same ID' >> beam.GroupBy(lambda m: m.id)
+            # Discards model.id from the PCollection.
+            | 'Get rid of key' >> beam.Values() # pylint: disable=no-value-for-parameter
+            # Only keep groupings that indicate that
+            # the UserStatsModel is missing.
+            | 'Filter pairs of models' >> beam.Filter(
+                lambda models: len(list(models)) == 1)
```

Comment:
Should we add protection against cases where the UserSettingsModel is the one that's missing?

Reply:
Done.